### PR TITLE
Improvements: Normalize users config

### PR DESCRIPTION
**Pull request by lciolecki**

### CODE REVIEW

File: docs/Security.md [hunk 1]
```diff
@@ -31,20 +31,33 @@ authentication: {
...
 users: [
   {
-    id: "user1"
-    password: "pass"
-    categoryPermissions: {
-      "Category1": ["Read", "Deploy"]
-      "Category2": ["Read", "Write"]
-    }
-    globalPermissions: ["AdminTab"]
+    identity: "user1"
```
Comment (arkadius) [inline docs/Security.md]: I think that we need to start updating `BreakingChanges.md`...

File: engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/RulesSet.scala [hunk 2]
```diff
@@ -35,8 +34,10 @@ class RulesSet(rules: List[OAuth2ConfigRule], allCategories: List[String]) {
 }
 
 object RulesSet {
-  def getOnlyMatchingRules(roles: List[String], rules: List[OAuth2ConfigRule], allCategories: List[String]): RulesSet = {
+  def getOnlyMatchingRules(roles: List[String], rules: List[ConfigRule], allCategories: List[String]): RulesSet = {
     val filtered = rules.filter(rule => roles.map(_.toLowerCase).contains(rule.role.toLowerCase))
     new RulesSet(filtered, allCategories)
   }
+
+  def apply(rules: List[ConfigRule], allCategories: List[String]): RulesSet = new RulesSet(rules, allCategories)
```
Comment (arkadius) [inline engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/RulesSet.scala]: Is it needed? `getOnlyMatchingRules` is not the only usage? There u use constructor.

File: docs/Security.md (hunk 1 above)
Comment (lciolecki) [reply docs/Security.md]: In future.. :)

File: engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/RulesSet.scala (hunk 2 above)
Comment (lciolecki) [reply engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/RulesSet.scala]: You'r right - removed.

**Pull request merged**